libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,134 @@
+        let mut images: Vec<DynamicImage> = Vec::new();
+
+        for page in pdf.pages().iter() {
+            //may adjust resolution depending on need
+            let resolution_width = 1000;
+
+            // Render the entire page to an image
+            let rendered_page = page
+                .render_with_config(&PdfRenderConfig::new().set_target_width(resolution_width))
+                .change_context(ChonkyError::Pdfium)?; // Renders the page to a PdfBitmap
+
+            // Convert PdfBitmap to DynamicImage
+            let dynamic_image = rendered_page.as_image();
+            images.push(dynamic_image);
+        }
+
+        Ok(images)
+    }
 }
 
 #[cfg(test)]
 mod tests {
+    use error_stack::{Report, ResultExt};
+    use pdfium_render::prelude::*;
+
     use super::*;
 
     #[test]
-    fn it_works() {
-        let result = add(2, 2);
-        assert_eq!(result, 4);
+    fn pdf_load_success() -> Result<(), Report<ChonkyError>> {
+        let pdfium = Pdfium::new(
+            Pdfium::bind_to_library(Pdfium::pdfium_platform_library_name_at_path("./libs/"))
+                .change_context(ChonkyError::Pdfium)?,
+        );
+
+        let test_pdf_string = "tests/docs/test-doc.pdf";
+
+        let _pdf = pdf_segmentation::load_pdf(&pdfium, test_pdf_string)
+            .change_context(ChonkyError::Pdfium)?;
+
+        Ok(())
+    }
+
+    #[test]
+    fn pdf_load_failure() -> Result<(), Report<ChonkyError>> {
+        let pdfium = Pdfium::new(
+            Pdfium::bind_to_library(Pdfium::pdfium_platform_library_name_at_path("./libs/"))
+                .change_context(ChonkyError::Pdfium)?,
+        );
+
+        let test_pdf_string = "tests/docs/invalid.pdf";
+
+        // Should return an error when loading an invalid PDF
+        let result = pdf_segmentation::load_pdf(&pdfium, test_pdf_string)
+            .change_context(ChonkyError::Pdfium);
+
+        if result.is_err() {
+            // Expected failure, return Ok
+            Ok(())
+        } else {
+            // Unexpected success, return an error
+            Err(Report::new(ChonkyError::Pdfium).attach_printable("Expected load_pdf to fail"))
+        }
+    }
+
+    #[test]
+    fn pdf_image_conversion() -> Result<(), Report<ChonkyError>> {
+        //create the pdfium instance and bind to library
+        let pdfium = Pdfium::new(
+            Pdfium::bind_to_library(Pdfium::pdfium_platform_library_name_at_path("./libs/"))
+                .change_context(ChonkyError::Pdfium)?,
+        );
+
+        let test_pdf_string = "tests/docs/test-doc.pdf";
+
+        let pdf = pdf_segmentation::load_pdf(&pdfium, test_pdf_string)
+            .change_context(ChonkyError::Pdfium)?;
+
+        let preprocessed_pdf =
+            pdf_segmentation::pdf_to_images(&pdf).change_context(ChonkyError::Pdfium)?;
+
+        let num_pages: usize = pdf.pages().len().into(); //number of pages of pdf
+
+        match preprocessed_pdf.len() {
```

Comment:
this isn't really a pressing issue, but it seems to me that this is only testing that `pdfium` converts a document into a number of pages - but we won't know what it'll look like. Maybe it'd make sense to take a smaller PDF (like 2 pages) and then use a kind of snapshot test - where we save the expected image output and the compare them. This way we're able to ensure that conversion actually takes place and that said conversion is consistent.

> as a side-note as it may be of interest - the snapshotting library we usually use, insta, has added experimental support for binary snapshots: https://github.com/mitsuhiko/insta/pull/610, while not really needed here - it might be interesting to see if it'd work here. Libraries like these allow us to more easily update snapshot tests, which is why I tend to use them.

Reply:
Yes, I'll look into some of the examples they have and try to implement this now especially if the library will be used in the rest of the package.